File: 3622-check-divisibility-by-digit-sum-and-product/3622-check-divisibility-by-digit-sum-and-product.py
class Solution:
    def checkDivisibility(self, n: int) -> bool:
        orig = n
        sum = 0
        temp = n
        prod = 1
        while(temp!=0):
           last_digit = temp%10  
           sum  = sum+last_digit
           prod = prod*last_digit
           temp = temp//10  
        ans = sum+prod
        if ans == 0:
            return false
        return orig%ans==0
            

File: 3622-check-divisibility-by-digit-sum-and-product/3622-check-divisibility-by-digit-sum-and-product.java
class Solution {
    public boolean checkDivisibility(int n) {
        int orig = n;
        int sum = 0;
        int temp = n;
        while(temp!=0){ 
           sum  = sum+(temp%10);
           temp = temp/10;
        }
        int prod = 1;
        temp = n;  
        while(temp!=0){
           prod = prod*(temp%10);
           temp=temp/10;
        }
        int ans = sum+prod;
        return orig%ans==0;
            
    }
}